include/proteus/Frontend/Func.hpp

```diff
@@ -207,12 +213,15 @@ class FuncBase {
   std::enable_if_t<std::is_arithmetic_v<T>, Var<T>>
   atomicMin(const Var<T *> &Addr, const Var<T> &Val);
 
-  template <typename T, typename BodyLambda = EmptyLambda>
-  auto forLoop(std::initializer_list<Var<T>> Bounds, BodyLambda &&Body = {}) {
-    auto It = Bounds.begin();
-    LoopBoundInfo<T> BoundsInfo{It[0], It[1], It[2], It[3]};
-    return ForLoopBuilder<T, BodyLambda>(BoundsInfo, *this,
-                                         std::forward<BodyLambda>(Body));
+  template <typename IterT, typename InitT, typename UpperT, typename IncT,
+            typename BodyLambda = EmptyLambda>
+  auto forLoop(Var<IterT> &Iter, const Var<InitT> &Init,
+               const Var<UpperT> &Upper, const Var<IncT> &Inc,
+               BodyLambda &&Body = {}) {
+    using RawT = std::remove_const_t<IterT>;
```

Comment:
I certainly like this interface more. Do we have to create `RawT`? Shouldn't we error out instead if the provided iterator is specified `const`?